src/helm/proxy/clients/http_client.py

```diff
@@ -0,0 +1,151 @@
+from dataclasses import asdict
+
+from helm.common.cache import Cache, CacheConfig
+from helm.common.request import (
+    Request,
+    RequestResult,
+    Sequence,
+    Token,
+    EMBEDDING_UNAVAILABLE_REQUEST_RESULT,
+)
+from helm.common.tokenization_request import (
+    DecodeRequest,
+    DecodeRequestResult,
+    TokenizationRequest,
+    TokenizationRequestResult,
+    TokenizationToken,
+)
+from .client import Client, wrap_request_time
+
+import requests
+
+
+class HTTPClient(Client):
+    """Implements a simple HTTP client."""
+
+    def __init__(
+        self,
+        cache_config: CacheConfig,
+        base_url: str = "http://localhost",
+        port: int = 8080,
+        timeout: int = 10,
+        do_cache: bool = False,
+    ):
+        self.cache = Cache(cache_config)
```

Comment:
```
self.cache: Optional[Cache] = Cache(cache_config) if do_cache else None
```
then you skip instantiating the cache if you don't need it. Then you can also delete all uses of `self.do_cache`.